lib/src/over_react_test/common_component_util.dart

```diff
@@ -496,24 +496,67 @@ void testRequiredProps(BuilderOnlyUiFactory factory, dynamic childrenFactory(),
         consoleErrors = [];
         PropTypes.resetWarningCache();
       });
-
-      addTearDown(() => context['console']['error'] = originalConsoleError);
     });
   }
 
   test('nullable props', () {
-    nullableProps.forEach((String propKey) {
-      var badRenderer = () => render((factory()..remove(propKey)(childrenFactory())));
+    if (!isComponent2) {
+      nullableProps.forEach((String propKey) {
+        var badRenderer = () => render((factory()..remove(propKey))(childrenFactory()));
 
-      expect(badRenderer, throwsPropError_Required(propKey, keyToErrorMessage[propKey]), reason: 'should throw when the required, nullable prop $propKey is not set');
+        expect(badRenderer, throwsPropError_Required(propKey, keyToErrorMessage[propKey]), reason: 'should throw when the required, nullable prop $propKey is not set');
 
-      var propsToAdd = {propKey: null};
-      badRenderer = () => render((factory()
-        ..addAll(propsToAdd)
-      )(childrenFactory()));
+        var propsToAdd = {propKey: null};
+        badRenderer = () => render((factory()
+          ..addAll(propsToAdd)
+        )(childrenFactory()));
 
-      expect(badRenderer, returnsNormally, reason: 'does not throw when the required, nullable prop $propKey is set to null');
-    });
+        expect(badRenderer, returnsNormally, reason: 'does not throw when the required, nullable prop $propKey is set to null');
+      });
+    } else if (isComponent2) {
+      PropTypes.resetWarningCache();
+
+      List<String> consoleErrors = [];
+      JsFunction originalConsoleError = context['console']['error'];
+      addTearDown(() => context['console']['error'] = originalConsoleError);
+      context['console']['error'] = new JsFunction.withThis((self, [message, arg1, arg2, arg3,  arg4, arg5]) {
+        consoleErrors.add(message);
+        originalConsoleError.apply([message, arg1, arg2, arg3,  arg4, arg5],
+            thisArg: self);
+      });
+
+      final reactComponentFactory = factory().componentFactory as
+      ReactDartComponentFactoryProxy2; // ignore: avoid_as
+
+      nullableProps.forEach((String propKey) {
+        if (!reactComponentFactory.defaultProps.containsKey(propKey)) {
```

Comment:
Can we add a test case for this? Neither `TestCommonRequired` nor `TestCommonRequired2` have any default props.

Reply:
added a default prop to both :)

Reply:
Should those defaults be `nullableRequiredProp` instead?

Reply:
derp